src/python-common/ceph/deployment/service_spec.py

```diff
@@ -830,6 +832,7 @@ def _cls(cls: Type[ServiceSpecT], service_type: str) -> Type[ServiceSpecT]:
             'jaeger-collector': TracingSpec,
             'jaeger-query': TracingSpec,
             'jaeger-tracing': TracingSpec,
+            SMBSpec.service_type: SMBSpec,
```

Comment:
In abstract, I always prefer to use a constant var / class var / enum over a plain string because then the language and static checkers can help find typos. Of course, like everyone does somtimes I get lazy ;-).